src/app/zeko/da_layer/README.md

```diff
@@ -0,0 +1,53 @@
+# Data Availability Layer
+
+## Node Description
+
+The data availability multisig, as explained here #12, is a very simple idea and previous implementation was extremely complex. We don't need EVM nor the blockchain for executing the simple multisig architecture. In the new simplified solution of DA committee, each node is a standalone service, which on-request stores and signs the batch. The signature is an attestation that the node knows and has the information on how to reproduce the ledger.
+
+The interface of the node API is roughly the following (`t` is the DA data unit):
+
+```ocaml
+type t =
+      { source_ledger_hash : Ledger_hash.t
+      ; target_ledger_hash : Ledger_hash.t
+      ; diff : (int * Account.t) list
+      ; command_with_action_step_flags :
+          (User_command.t * bool list) option
+      }
+
+val post_batch : ledger_openings:Sparse_ledger.t -> batch:t-> Signature.t
+val get_batch : Ledger_hash.t -> t option
+```
+
+Each data batch has the all the account changes stored in the `diff`. The da node applies each account's diff to the ledger openings, and compares the resulting ledger hash with the `target_ledger_hash`. If the adversary would want to lie, and therefore convince node to sign the ledger hash that the node doesn't know how to construct, the diff would need to be either incorrect, or some would be missing. To ensure neither happens following steps are taken:
+
+1. Check that `root ledger_openings = batch.source_ledger_hash`.
+2. Check that `batch.source_ledger_hash` is either in the database or an empty ledger.
+3. Check that the indices in `batch.diff` are unique.
+4. Set each account in `batch.diff` to the `ledger_openings` and call the resulting ledger root `target_ledger_hash`.
+5. Sign `target_ledger_hash`.
+6. Store the batch under the `target_ledger_hash`.
+
+For convenience, we store also the full command, which helps to not only reconstruct the ledger, but also have to full history of the state. To ensure that the command is a correct one we do one additional step:
+
+- Check that after applying all the receipts of the command, the receipt chain hashes match the target ledger.
+
+The receipt chain hash is a commitment to all the transaction commitments. Transaction commitment doesn't include authorization fields, so the adversary can lie about the authorization of the corresponding command, this however doesn't allow to convince the da node to sign the incorrect ledger hash.
```

Comment:
But account update includes the information?

Reply:
What is meant here: DA doesn’t know if authorization was provided, but this isn’t relevant to DA.